Quick/DSL/World+DSL.swift

```diff
@@ -17,8 +17,11 @@ extension World {
     }
 
     internal func describe(description: String, flags: FilterFlags, closure: () -> ()) {
+        guard currentExampleGroup != nil else {
+            fatalError("Error: example group was not created by its parent QuickSpec spec. Check that describe() was used in QuickSpec.spec() and not a more general context (i.e. an XCTestCase test)")
+        }
```

Comment:
This is a great improvement! Thanks! :100: 

Two suggestions:

1. The end user may be using `context`, which simply calls `describe`. This error message could be updated to read "Check that context() or describe() was used in...", or something similar.
2. Would it be possible to add a unit test for this? I'm aware that the previous behavior wasn't unit tested either, but adding a test here would be a great way we catch regressions that would prevent users from seeing this helpful message. This may involve using an `NSException` instead of `fatalError`, since the latter cannot be unit tested (yet). I realize that's less "Swift-y", but from an end user's perspective, that shouldn't matter.

Thoughts? :raised_hands: